l1-contracts/src/mock/TestERC20.sol

```diff
@@ -7,27 +7,46 @@ import {ERC20} from "@oz/token/ERC20/ERC20.sol";
 import {IMintableERC20} from "./../governance/interfaces/IMintableERC20.sol";
 
 contract TestERC20 is ERC20, IMintableERC20, Ownable {
-  bool public freeForAll = false;
+  mapping(address => bool) public minters;
 
-  modifier ownerOrFreeForAll() {
-    if (msg.sender != owner() && !freeForAll) {
-      revert("Not owner or free for all");
-    }
+  event MinterAdded(address indexed minter);
+  event MinterRemoved(address indexed minter);
+
+  error NotMinter(address caller);
+
+  modifier onlyMinter() {
+    require(minters[msg.sender], NotMinter(msg.sender));
     _;
   }
 
   constructor(string memory _name, string memory _symbol, address _owner)
     ERC20(_name, _symbol)
     Ownable(_owner)
-  {}
-
-  // solhint-disable-next-line comprehensive-interface
-  function setFreeForAll(bool _freeForAll) external onlyOwner {
-    freeForAll = _freeForAll;
+  {
+    minters[_owner] = true;
   }
 
-  function mint(address _to, uint256 _amount) external override(IMintableERC20) ownerOrFreeForAll {
+  function mint(address _to, uint256 _amount) external override(IMintableERC20) onlyMinter {
     _mint(_to, _amount);
   }
+
+  function addMinter(address _minter) external override(IMintableERC20) onlyOwner {
+    minters[_minter] = true;
+    emit MinterAdded(_minter);
+  }
+
+  function removeMinter(address _minter) external override(IMintableERC20) onlyOwner {
+    minters[_minter] = false;
+    emit MinterRemoved(_minter);
+  }
+
+  function transferOwnership(address newOwner) public override(Ownable) onlyOwner {
+    if (newOwner == address(0)) {
+      revert OwnableInvalidOwner(address(0));
+    }
+    minters[owner()] = false;
+    _transferOwnership(newOwner);
+    minters[newOwner] = true;
```

Comment:
Slightly annoying here, but this is not emitting the events that you would probably expect, simplest might be to make the add and remove public and call those directly.